cf/samples/javascript-samples/geotab/geotab-to-sap-cloud.js

```diff
@@ -0,0 +1,768 @@
+ * @return {Object}  the parsed configuration file
+ */
+function readConfigFile(filename) {
+    const abs_path = path.join(__dirname, filename);
+    const json = fs.readFileSync(abs_path, {
+        'encoding': 'utf8'
+    });
+    return JSON.parse(json);
+}
+
+
+/**
+ * Create the SAP router device unless it already exists and fetch its certificate.
+ * @return {Object} A proxy object with postMeasurementData() asynchronous method.
+ */
+async function initRouterDevice() {
+    const routerDeviceId = await createDeviceUnlessExists(gatewayId, 'router', true);
+    const absPathPem = path.join(__dirname, config.SAP_router_device_pem_filename);
+    const absPathPass = path.join(__dirname, config.SAP_router_device_pass_filename);
+    if (!(fs.existsSync(absPathPem) && fs.existsSync(absPathPass))) {
+        const resp = await SAP_DeviceManagement.apis.Devices.getPEMCertificateUsingGET({
+            tenantId: config.SAP_tenantID,
+            deviceId: routerDeviceId
+        });
+        // console.debug(resp);
+        if (resp.ok) {
+            fs.writeFileSync(absPathPem, resp.body.pem);
+            fs.writeFileSync(absPathPass, resp.body.secret);
+            console.info('Saved the PEM and the passphrase for the router device.');
+        }
+    }
+
+    const router_device_pem = fs.readFileSync(absPathPem, {
+        'encoding': 'utf8'
+    }); // contains both the certificate and the private key
+    const router_device_passphrase = fs.readFileSync(absPathPass, {
+        'encoding': 'utf8'
+    }); // the private key is encrypted with it
+    console.debug('The PEM and the passphrase for the router device read OK.');
+    const gateway_hostname = `${config.SAP_instance}.${config.SAP_landscape}`;
+    return new SAP_IoT_device_REST(gateway_hostname, router_device_pem, router_device_pem, router_device_passphrase);
+}
+
+/** 
+ * Create the SAP IoT data type to which the data of Geotab typeName 'LogRecord' are mapped. 
+ *
+ * The data type is specified according to the SAP IoT Service rules as a Capability and a Sensor Type
+ * providing the corresponding measurement data.
+ */
+async function createType_LogRecord() {
+    const capId = await createCapabilityUnlessExists('LogRecord', [
+        { name: 'latitude',  dataType: DataType.FLOAT },
+        { name: 'longitude', dataType: DataType.FLOAT },
+        { name: 'speed',     dataType: DataType.FLOAT },
+        { name: 'dateTime',  dataType: DataType.DATE }
+    ]);
+    return createSensorTypeUnlessExists('LogRecord', capId);
+}
+
+/** 
+ * Map the Geotab typeName 'StatusData' and the 'Diagnostic' it references to SAP IoT Service data model. 
+ *
+ * The data type is specified according to the SAP IoT Service rules as a Capability and a Sensor Type
+ * providing the corresponding measurement data.
+ */
+async function createType_StatusData() {
+    const capId = await createCapabilityUnlessExists('StatusData', [
```

Comment:
as general suggestion (personal opinion) put the comment always on a new line because it could break the line formatting (related to format max font per line)